Linker cannot find "boost_python-vc140-mt-x64-1_66.lib" on Windows.

This relates directly to #572.

So first it seems like irrespective of whether I use `-DOPENEXR_BUILD_PYTHON_LIBS=0`, the Python bindings are being built and then I get a few linker errors:

```
LINK : fatal error LNK1104: cannot open file 'boost_python-vc140-mt-x64-1_66.lib' [***\openexr\build\platform-windows\arch-AMD64\openexr\PyIlmBase\PyIex\iex_python2.vcxproj]
[...]
LINK : fatal error LNK1104: cannot open file 'boost_python-vc140-mt-x64-1_66.lib' [***\openexr\build\platform-windows\arch-AMD64\openexr\PyIlmBase\PyImath\imath_python2.vcxproj]
```

I'm not sure where it tries to get the library from but I can see it here along with the shared version and a name variation:
- ***\boost\1.66.0.rf1\platform-windows\arch-AMD64\python-2.7.14\boost\stage\lib\boost_python-vc140-mt-x64-1_66.lib
- ***\boost\1.66.0.rf1\platform-windows\arch-AMD64\python-2.7.14\boost\stage\lib\boost_python-vc140-mt-x64-1_66.dll
- ***\boost\1.66.0.rf1\platform-windows\arch-AMD64\python-2.7.14\boost\stage\lib\libboost_python-vc140-mt-x64-1_66.lib

Here is a doctored configuration log if that helps:

```
-- Selecting Windows SDK version  to target Windows 10.0.17763.
-- The C compiler identification is MSVC 19.0.24210.0
-- The CXX compiler identification is MSVC 19.0.24210.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configure ILMBASE Version: 2.4.0 Lib API: 24.0.0
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Looking for include file ucontext.h
-- Looking for include file ucontext.h - not found
NOTICE-- WARNING pkg-config generation disabled
-- Configure OpenEXR Version: 2.4.0 Lib API: 24.0.0
CMake Warning (dev) at CMakeLists.txt:29 (_find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable ZLIB_ROOT is set to:

    ***\zlib\1.2.11.rf1\platform-windows\arch-AMD64\zlib

  For compatibility, CMake is ignoring the variable.
Call Stack (most recent call first):
  OpenEXR/config/OpenEXRSetup.cmake:136 (find_package)
  OpenEXR/CMakeLists.txt:30 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found ZLIB: ***/zlib/1.2.11.rf1/platform-windows/arch-AMD64/zlib/lib/zlib.lib (found version "1.2.11")
-- Performing Test OPENEXR_IMF_HAVE_SYSCONF_NPROCESSORS_ONLN
-- Performing Test OPENEXR_IMF_HAVE_SYSCONF_NPROCESSORS_ONLN - Failed
-- Performing Test OPENEXR_IMF_HAVE_GCC_INLINE_ASM_AVX
-- Performing Test OPENEXR_IMF_HAVE_GCC_INLINE_ASM_AVX - Failed
-- Configure PyIlmBase Version: 2.4.0 Lib API: 24.0.2
CMake Warning (dev) at CMakeLists.txt:29 (_find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Environment variable Python_ROOT is set to:

    C:\Python27

  For compatibility, CMake is ignoring the variable.
Call Stack (most recent call first):
  PyIlmBase/CMakeLists.txt:34 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Python: C:/Python27/python.exe (found version "2.7.14") found components:  Interpreter Development
-- Found Python2: C:/Python27/python.exe (found version "2.7.14") found components:  Interpreter Development
-- Could NOT find Python3: Found unsuitable major version "2", but required major version is exact version "3"
NOTICE: Found Python 2.7.14
-- Found Python2 libraries: 27
-- Boost version: 1.66.0
-- Found the following Boost libraries:
--   python
CMake Warning at PyIlmBase/CMakeLists.txt:104 (message):
  Ambiguous boost python module found, assuming python 2


-- Found NumPy 1.13.3 for Python 2.7.14: ***/numpy/1.13.3/numpy/Lib/site-packages/numpy/core/include
-- Configure OpenEXR Viewers Version: 2.4.0 Lib API: 24.0.1
-- Found OpenGL: opengl32
-- Could NOT find FLTK (missing: FLTK_LIBRARIES FLTK_INCLUDE_DIR)
CMake Warning at OpenEXR_Viewers/CMakeLists.txt:36 (message):
  FLTK not found, exrdisplay will not be built


NOTICE: Disabling playexr, not yet ported to WIN32 platforms
-- Configuring done
-- Generating done
-- Build files have been written to: ***/openexr/build/platform-windows/arch-AMD64/openexr
```

--- a/PyIlmBase/config/ModuleDefine.cmake
+++ b/PyIlmBase/config/ModuleDefine.cmake
@@ -116,6 +116,7 @@ function(PYILMBASE_DEFINE_MODULE modname)
         ${extraDeps}
         ${PYILMBASE_CURMOD_DEPENDENCIES}
         ${PYILMBASE_CURMOD_PRIVATE_DEPS}
+        Boost::${PYILMBASE_BOOST_PY2_COMPONENT}
       )
     set_target_properties(${modname}_python2 PROPERTIES
       LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/python${Python2_VERSION_MAJOR}_${Python2_VERSION_MINOR}/"
@@ -142,6 +143,7 @@ function(PYILMBASE_DEFINE_MODULE modname)
         ${libname} ${extraDeps}
         ${PYILMBASE_CURMOD_DEPENDENCIES}
         ${PYILMBASE_CURMOD_PRIVATE_DEPS}
+        Boost::${PYILMBASE_BOOST_PY3_COMPONENT}
       )
     set_target_properties(${modname}_python3 PROPERTIES
       LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/python${Python3_VERSION_MAJOR}_${Python3_VERSION_MINOR}/"

